src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useContext, useState, useEffect, useRef } from 'react';
+import { Todo } from '../Todo/Todo';
+import { Notification } from '../Notification/Notification';
+import { TodoContext } from '../../contexts/TodoContext';
+import { USER_ID, createTodo, deleteTodo, updateTodo } from '../../api/todos';
+import {
+  FilterContext,
+  ACTIONSINFILTERCONTEXT,
+} from '../../contexts/FilterContext';
+import { Filter } from '../Filter/Filter';
+
+export const TodoApp: React.FC = () => {
+  const { todosFromServer, refreshTodos, putErrorWarning } =
+    useContext(TodoContext)!;
+  const {
+    filteredTodos,
+    currentFilterCriteria,
+    itemsLeft,
+    totalCompletedTodos,
+  } = useContext(FilterContext)!;
+  const [showFooterOrNot, setShowFooterOrNot] = useState(false);
+  const [textInFormToCreateTodo, settextInFormToCreateTodo] = useState('');
+  const refInputCreatTodo = useRef<HTMLInputElement>(null);
+
+  const hanldeInputValueFormCreateTodo = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    settextInFormToCreateTodo(event.target.value);
+  };
+
+  useEffect(() => {
+    refInputCreatTodo.current!.focus();
```

Comment:
It is best to use null-checker instead of non-null assertion, because this CAN be null.
`refInputCreatTodo.current?.focus();`

Reply:
In fact in that case is certain that is not-null, in this run time execution refInputCreateTodo is assert to tsx element already render in return.